activesupport/lib/active_support/testing/assertions.rb

```diff
@@ -276,6 +279,15 @@ def _assert_nothing_raised_or_warn(assertion, &block)
 
           raise
         end
+
+        def _callable_to_source_string(callable)
+          if defined?(RubyVM) && callable.is_a?(Proc)
+            ast = RubyVM::AbstractSyntaxTree.of(callable, keep_script_lines: true)
+            "-> #{ast.source.strip}"
```

Comment:
Could we use [Prism](https://github.com/ruby/prism) here? I wonder if that would help with other Ruby implementations

Reply:
Hm I think Prism does not have an API to get the source of a method. The only thing I can think of is parsing the file with Prism and trying to find the location of the assertion (probably by using the line number from `caller`?). 

Maybe there is another way I do not know of, but right know I feel like it would be way too complicated to achieve this using Prism.